### DISCUSSION TITLE
Copilot.vim core dumps

### DISCUSSION THREAD

**Discussion by macbuse** [community/community#37867] [Copilot Conversations] [answered]
nvim/Copilot.vim core dumps  with 

free(): invalid pointer 

on  \<CR\> to select solution as per help :Copilot

---
Node   v17.3.0

NVIM v0.9.0-dev                                                                                                                                                              
Build type: RelWithDebInfo                                                                                                                                                   
LuaJIT 2.1.0-beta3       

NAME="Ubuntu"
VERSION="18.04.6 LTS

**Comment by iamcjwilliams** [comment]
@macbuse Our copilot team is investigating this

**Comment by tpope** [comment, marked as answer]
Neovim crashing is generally a Neovim bug. I'd encourage you to try the latest stable release—0.8.0—and see if the problem persists.

**Reply by macbuse** [reply]
Hi

thanks for getting back to me. I figured that it's the neovim code.

I reverted to:

NVIM v0.7.0                                                                                                                                                                  
Build type: RelWithDebInfo                                                                                                                                                   
LuaJIT 2.1.0-beta3        

but it still gives me the core dump

It's a standard error on invoking free() after malloc in C (I've had it for 25+ years)
therefore in neovim and not copilot.vim. I don't get a trace on a core dump 
so I wanted to know what your callback was on the  \<CR\> that I was doing  
to post it as an error for the maintainer(s).

sorry

**Reply by tpope** [reply]
Basically, it pulls the solution out of the panel and inserts it into the original buffer using `setbufline()` and `appendbufline()`. The exact code is here:

https://github.com/github/copilot.vim/blob/8ba151a20bc1d7a5c72e592e51bfc925d5bbb837/autoload/copilot/panel.vim#L66-L102

**Comment by macbuse** [comment]
Hi

Thanks, that's great. Sorry for bothering you but I'm
using Copilot  a lot and well it's fun when it works :)
So u can mark this as closed.

I crossposted https://github.com/neovim/neovim/issues/20922
 and it's probably (?)  a pb with lib.so.6 once I figured how to get the
trace:

Message: Process 19728 (nvim) of user 1001 dumped core. Stack trace of
thread 19728: #0 0x00007fc137702e87 raise (libc.so.6) #1
<https://github.com/neovim/neovim/pull/1> 0x00007fc1377047f1 abort
(libc.so.6) #2 <https://github.com/neovim/neovim/pull/2> 0x00007fc13774d837
n/a (libc.so.6) #3 <https://github.com/neovim/neovim/issues/3>
0x00007fc1377548ba n/a (libc.so.6) #4
<https://github.com/neovim/neovim/issues/4> 0x00007fc13775bdec cfree
(libc.so.6) #5 <https://github.com/neovim/neovim/pull/5> 0x0000564c319f83eb
n/a

On Thu, 3 Nov 2022 at 22:22, Tim Pope ***@***.***> wrote:

> Basically, it pulls the solution out of the panel and inserts it into the
> original buffer using setbufline() and appendbufline(). The exact code is
> here:
>
>
> https://github.com/github/copilot.vim/blob/8ba151a20bc1d7a5c72e592e51bfc925d5bbb837/autoload/copilot/panel.vim#L66-L102
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/37867#discussioncomment-4051333>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELF7CL534ZTC7N6AC54JL3WGQUJXANCNFSM6AAAAAARS3TTQI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
greg mc

Grenoble France

https://www-fourier.ujf-grenoble.fr/~mcshane/
